ansible/roles/pmix/tasks/main.yaml

```diff
@@ -0,0 +1,44 @@
+---
+
+- name: Include OS Family Dependent Vars
+  ansible.builtin.include_vars: '{{ item }}'
+  with_first_found:
+  - files:
+    - '{{ ansible_distribution|lower }}-{{ ansible_distribution_version }}.yml'
+    - '{{ ansible_distribution|lower }}-{{ ansible_distribution_major_version }}.yml'
+    - '{{ ansible_distribution|lower }}.yml'
+    - '{{ ansible_os_family|lower }}-{{ ansible_distribution_version }}.yml'
+    - '{{ ansible_os_family|lower }}-{{ ansible_distribution_major_version }}.yml'
+    - '{{ ansible_os_family|lower }}.yml'
+
+- name: Wait for DPKG Locks
+  ansible.builtin.shell: >
+    while fuser /var/lib/dpkg/{{ item }} >/dev/null 2>&1; do
+      sleep 5
+    done
+  with_items:
+  - lock
+  - lock-frontend
+  when:
+  - ansible_os_family == 'Debian'
+
+- name: Install PMIx libraries and headers
+  ansible.builtin.package:
+    name: '{{ pmix_dependencies }}'
+    state: present
+
+- name: Download PMIX
+  ansible.builtin.get_url:
+    url: https://github.com/openpmix/openpmix/releases/download/{{ pmix_version }}/pmix-{{
+      pmix_version }}.tar.gz
+    dest: /tmp/pmix-{{ pmix_version }}.tar.gz
+    checksum: sha1:912c70f567098dd0a7c0a03a816c5e6ecfaf1962
+
+- name: Build PMIx
+  ansible.builtin.shell: |
```

Comment:
please add a [creates](https://docs.ansible.com/ansible/2.9/modules/shell_module.html#examples) flag here that blocks on a re-run. e.g. the lib.so file that gets created in /usr/local. I realize this Ansible playbook isn't very idempotent already, but we should not add to the problem.

Reply:
Done